### PR TITLE
Routing

### LINKED ISSUE #301: Integrate stack-based react-native router
Try using react-navigation in react-native-web. See: https://github.com/react-navigation/react-navigation/issues/2298

### PR DESCRIPTION
**Pull request by tanx**
Closes #301

### CODE REVIEW

File: src/view/channel-create.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ const ChannelCreateView = ({ store, nav, channel }) => (
     <Header separator>
       <Button disabled onPress={() => {}} />
       <Title title="Create Channel" />
-      <CancelButton onPress={() => nav.goChannels()} />
+      <CancelButton onPress={() => nav.goBack()} />
```
Comment (valentinewallace) [inline src/view/channel-create.js]: that's a nice feature, feel like it eliminates a class of potential mistakes :+1:

File: src/view/home.js [hunk 2]
```diff
@@ -53,7 +53,7 @@ const HomeView = ({
         goSettings={() => nav.goSettings()}
         showChannelAlert={store.showChannelAlert}
       />
-      <QrCodeSeparator goFundWallet={() => nav.goFundWallet()} />
+      <QrCodeSeparator goDeposit={() => nav.goDeposit()} />
```
Comment (valentinewallace) [inline src/view/home.js]: makes more sense

File: src/navigator.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+} from './action';
+import store from './store';
+
+const history = syncHistoryWithStore(createBrowserHistory(), store.route);
+
+class Navigator extends Component {
+  render() {
+    const { lastNotification, displayNotification } = store;
+    return (
+      <Router history={history}>
```
Comment (valentinewallace) [inline src/navigator.js]: Nice how this enables the history feature & looks much nicer than the previous `main.js`

File: test/unit/action/nav.spec.js [hunk 4]
```diff
@@ -25,211 +28,251 @@ describe('Action Nav Unit Tests', () => {
 
   describe('constructor()', () => {
     it('should listen to open-url event', () => {
-      expect(nav._store, 'to be ok');
+      expect(store.route.location.pathname, 'to equal', '/');
+      expect(store.route.history.entries.length, 'to equal', 1);
```
Comment (valentinewallace) [inline test/unit/action/nav.spec.js]: so convenient :) i see why you wanted to integrate this library

**Comment by tanx** [comment]
> etc., pretty much missing all back buttons/exit buttons and the + sign in the Add button for the channels screen.

Good catch! I'll take a look.

I should actually try to integrate the iOS navigation into the architecture as well to get a generic solution. I'll close the PR for now.

**Pull request closed without merging**